src/dsp_tools/commands/create/create_on_server/properties.py

```diff
@@ -0,0 +1,119 @@
+from http import HTTPStatus
+from typing import Any
+
+import rustworkx as rx
+from loguru import logger
+from rdflib import Literal
+from rdflib import URIRef
+from tqdm import tqdm
+
+from dsp_tools.clients.ontology_clients import OntologyCreateClient
+from dsp_tools.commands.create.create_on_server.onto_utils import get_onto_lookup
+from dsp_tools.commands.create.create_on_server.onto_utils import should_retry_request
+from dsp_tools.commands.create.create_on_server.onto_utils import sort_for_upload
+from dsp_tools.commands.create.models.create_problems import CollectedProblems
+from dsp_tools.commands.create.models.create_problems import CreateProblem
+from dsp_tools.commands.create.models.create_problems import UploadProblem
+from dsp_tools.commands.create.models.create_problems import UploadProblemType
+from dsp_tools.commands.create.models.parsed_ontology import ParsedProperty
+from dsp_tools.commands.create.models.server_project_info import CreatedIriCollection
+from dsp_tools.commands.create.models.server_project_info import ListNameToIriLookup
+from dsp_tools.commands.create.models.server_project_info import OntoCreateLookup
+from dsp_tools.commands.create.models.server_project_info import ProjectIriLookup
+from dsp_tools.commands.create.serialisation.ontology import serialise_property_graph_for_request
+
+
+def create_all_properties(
+    properties: list[ParsedProperty],
+    project_iri_lookup: ProjectIriLookup,
+    created_iris: CreatedIriCollection,
+    list_lookup: ListNameToIriLookup,
+    client: OntologyCreateClient,
+) -> tuple[CreatedIriCollection, CollectedProblems | None]:
+    upload_order = _get_property_create_order(properties)
+    logger.debug(f"Property creation order: {upload_order}")
+    prop_lookup = {p.name: p for p in properties}
+    all_problems: list[CreateProblem] = []
+    onto_lookup = get_onto_lookup(project_iri_lookup, client)
+    progress_bar = tqdm(upload_order, desc="    Creating properties", dynamic_ncols=True)
+    logger.debug("Starting property creation")
+    for p in progress_bar:
+        prop = prop_lookup[p]
+        if previous_blocker := _is_property_blocked(prop, created_iris):
+            all_problems.append(previous_blocker)
+            created_iris.failed_properties.add(prop.name)
+        else:
+            list_iri = None
+            if prop.node_name is not None:
+                if not (found := list_lookup.get_iri(prop.node_name)):
+                    all_problems.append(UploadProblem(prop.name, UploadProblemType.PROPERTY_LIST_NOT_FOUND))
+                    continue
+                else:
+                    list_iri = Literal(f"hlist=<{found}>")
+            create_result = _create_one_property(prop, list_iri, onto_lookup, client)
+            if isinstance(create_result, Literal):
+                onto_lookup.add_last_mod_date(prop.onto_iri, create_result)
```

Comment:
I would rename that to `update_last_mod_date()`, because you don't add a new LMD. Rather you overwrite=update an existing one.